### PR TITLE
Migrate algod and indexer clients to TypeScript

### LINKED ISSUE #303: Migrate Algodv2 and Indexer clients to TypeScript
## Summary
This issue tracks the migration of the Algodv2 and Indexer clients to TypeScript.

## Scope
Migrate Algodv2 and Indexer clients, along with all their request types to TypeScript.

## Urgency
Required to finish the TypeScript migration.

### PR DESCRIPTION
**Pull request by jdtzmn**
This PR migrates the Algodv2 client, indexer client, and all of their request types to TypeScript.

Closes #303.

### CODE REVIEW

File: src/types/intDecoding.d.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ *     Number.MAX_SAFE_INTEGER will lose precision.
+ *   * "safe": All integers will be decoded as Numbers, but if any values are greater than
+ *     Number.MAX_SAFE_INTEGER an error will be thrown.
+ *   * "mixed": Integers will be decoded as Numbers if they are less than or equal to
+ *     Number.MAX_SAFE_INTEGER, otherwise they will be decoded as BigInts.
+ *   * "bigint": All integers will be decoded as BigInts.
+ *
+ *   Defaults to "default" if not included.
+ */
+export type IntDecoding = 'default' | 'safe' | 'mixed' | 'bigint';
```
Comment (jasonpaulos) [inline src/types/intDecoding.d.ts:15]: I think an enum might make more sense for this field, let me know if you disagree.

**Review by jasonpaulos — CHANGES_REQUESTED** [review]

File: src/client/v2/algod/supply.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import JSONRequest from '../jsonrequest';
+
+export default class Supply extends JSONRequest {
+  // eslint-disable-next-line no-underscore-dangle,class-methods-use-this
```
[the remark this answers is not in the document]
Comment (jdtzmn) [reply src/client/v2/algod/supply.ts:4]: Good catch!

File: src/client/v2/indexer/lookupApplications.ts [hunk 3]
```diff
@@ -17,5 +18,3 @@ class LookupApplications extends JSONRequest {
     return this;
```
[the remark this answers is not in the document]
Comment (jdtzmn) [reply src/client/v2/indexer/lookupApplications.ts:18]: Yes!

File: src/client/v2/indexer/lookupTransactionByID.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import JSONRequest from '../jsonrequest';
+import { HTTPClient } from '../../client';
+import { IntDecoding } from '../../../types/intDecoding';
+
+export default class LookupTransactionByID extends JSONRequest {
+  constructor(c: HTTPClient, intDecoding: IntDecoding, private txID) {
```
[the remark this answers is not in the document]
Comment (jdtzmn) [reply src/client/v2/indexer/lookupTransactionByID.ts:6]: Yes, good catch!

File: src/types/intDecoding.d.ts (hunk 1 above)
Comment (jdtzmn) [reply src/types/intDecoding.d.ts:15]: Yes, definitely. Initially I wasn't sure whether a union was needed for implementation reasons. I'll convert it to an enum.

**Review by jasonpaulos — APPROVED** [review]
Looks good!

**Pull request merged**